src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -628,29 +628,41 @@ private String formatColumnName(String columnName) {
    * @return the set of input column names
    */
   private Set<String> verifyInputColumns(Map<String, Object> row) {
-    Set<String> inputColumns =
-        row.keySet().stream().map(this::formatColumnName).collect(Collectors.toSet());
+    Map<String, String> inputColNamesMap =
+        row.keySet().stream().collect(Collectors.toMap(this::formatColumnName, value -> value));
 
-    for (String columnName : this.nonNullableFieldNames) {
-      if (!inputColumns.contains(columnName)) {
-        throw new SFException(
-            ErrorCode.INVALID_ROW,
-            "Missing column: " + columnName,
-            "Values for all non-nullable columns must be specified.");
+    // Check for extra columns in the row
+    List<String> extraCols = new ArrayList<>();
+    for (String columnName : inputColNamesMap.keySet()) {
+      Field field = this.fields.get(columnName);
+      if (field == null) {
+        extraCols.add(inputColNamesMap.get(columnName));
       }
     }
 
-    for (String columnName : inputColumns) {
-      Field field = this.fields.get(columnName);
-      if (field == null) {
-        throw new SFException(
-            ErrorCode.INVALID_ROW,
-            "Extra column: " + columnName,
-            "Columns not present in the table shouldn't be specified.");
+    if (!extraCols.isEmpty()) {
+      throw new SFException(
+          ErrorCode.INVALID_ROW,
+          "Extra columns: " + extraCols,
```

Comment:
Arrays.toString(extraCols)?

Reply:
I actually used the similar way (list.toString()), removing the toString is actually suggested by Intellij